src/containers/tutor-home-page/general-info-step/GeneralInfoStep.jsx

```diff
@@ -8,13 +8,23 @@ import AppTextArea from '~/components/app-text-area/AppTextArea'
 import AppTextField from '~/components/app-text-field/AppTextField'
 import AsyncAutocomplete from '~/components/async-autocomlete/AsyncAutocomplete'
 import { useStepContext } from '~/context/step-context'
+import { useAppSelector } from '~/hooks/use-redux'
 import { locationService } from '~/services/location-service'
 import img from '~/assets/img/tutor-home-page/become-tutor/general-info.svg'
 
 const GeneralInfoStep = ({ btnsBox, setIsValidated, stepLabel }) => {
   const { stepData, handleStepData } = useStepContext()
   const { t } = useTranslation()
 
+  const { userFirstName, userLastName } = useAppSelector(
+    (state) => state.appMain
+  )
+
+  useEffect(() => {
+    stepData.generalInfo.data.firstName = userFirstName
+    stepData.generalInfo.data.lastName = userLastName
+  }, [])
```

Comment:
Pay attention to the warning, and make sure these direct assignments update context as intended(if don't, use handleStepData instead).

Reply:
Yes, I agrre with @nadraharr. Use  `stepData.generalInfo.data.firstName` and `stepData.generalInfo.data.lastName` as initial data for the data in the context. Redux should save the same data as Back-end but stepper context shoud save you current typed name and surname.